### ISSUE TITLE
Lint when using PhantomData without lifetime

### ISSUE THREAD

**Issue by ThrashAbaddon** [rust-lang/rust-clippy#7618]
### What it does

Explains to the user that `PhantomData` field doesn't do anything because lifetime isn't available.

*What does this lint do?*

Informs the user that it has useless `PhantomData`  usage, and shows him the places where lifetime should be added.

### Categories (optional)

- Kind: *See <https://github.com/rust-lang/rust-clippy/blob/master/README.md#clippy> for list of lint kinds*

I would say `clippy::correctness` or `clippy::suspicious`

clippy::correctness
--

<br class="Apple-interchange-newline">
</body>
</html>

*What is the advantage of the recommended code over the original code*
- less code
- it is indicated that even though the code compiles fine there is a reason why `PhantomData` doesn't do anything.

### Drawbacks

None.

### Example

```rust
use std::marker::PhantomData;

pub struct Repository {}

impl Repository {
    fn create_commit(&self) -> Commit {
        Commit { _marker: PhantomData }
    }
}


pub struct Commit {
    _marker: PhantomData<Repository>,
}


#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_useless_phantom_data() {
        let repo = Repository {};
        let commit = repo.create_commit();

        drop(repo);
        drop(commit);
    }
}

```

Could be written as:

```rust
pub struct Commit<'a> {
    _marker: PhantomData<&'a Repository>,
}

```
